dojson.tpl

```diff
@@ -28,12 +28,50 @@ from ..model import marc21
 def {{ clean_name(field.name) }}(self, key, value):
     """{{ field.name }}."""
     {%- if indicator1.get('name') %}
+      {%- if indicator1.get('name') == 'nonfiling_characters' %}
+    indicator_map1 = {str(x): str(x) for x in range(10)}
+      {%- else %}
     indicator_map1 = {{ tojson(indicator1.get('values', {})) }}
+      {%- endif -%}
     {%- endif %}
     {%- if indicator2.get('name') %}
+      {%- if indicator2.get('name') == 'nonfiling_characters' %}
+    indicator_map2 = {str(x): str(x) for x in range(10)}
+      {%- else %}
     indicator_map2 = {{ tojson(indicator2.get('values', {})) }}
+      {%- endif -%}
     {%- endif %}
+
+    {%- set reverse_subfields = dict() %}
+    field_map = {
+      {%- for code, subfield in field.get('subfields').iteritems() %}
+        '{{ code }}': '{{ clean_name(subfield['name']) }}',
+        {%- do reverse_subfields.update({clean_name(subfield['name']): code}) %}
+      {%- endfor %}
+    }
+
+    order = utils.map_order(field_map, value)
+
+    {%- if indicator1.get('name') %}
+
+    if key[3] in indicator_map1
+      {%- if indicator1.get('name') in reverse_subfields.keys() -%}
```

Comment:
if you remove `-` from `-}}` then you don't need to do `{{ " " }}` on next line.